test/suites/api/test_object.c

```diff
@@ -780,6 +780,75 @@ static void test_bad_args(void) {
     json_decref(num);
 }
 
+static void test_keylen(void) {
+    json_t *obj = json_object();
+    const char key[] = {'t', 'e', 's', 't', '1'};
+    const char key2[] = {'t', 'e', 's', 't'};
```

Comment:
you might want to add a test case with a key containing a null, like:

```c
    const char key3[] = {'t', 'e', 's', '\0', 't'};
```

Reply:
Added, plus added sample:
const char key4[] = {'t', 'e', 's', 't', '\0'};